### PR TITLE
Add flag to only use cached complexes to avoid failures due to repeatedly failing to prep the structures

### PR DESCRIPTION
**Pull request by apayne97**
## Description
This made it possible to run the cross-docking runs to avoid the issue with https://github.com/choderalab/asapdiscovery/issues/695

## Status
- [x] Ready to go

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by apayne97** [comment]
@jthorton could you give this a quick review?

File: asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py [hunk 1]
```diff
@@ -122,6 +123,12 @@ def prep(
                 inputs, cached_outputs = ProteinPrepperBase._gather_new_tasks(
                     complex_to_prep=inputs, cached_complexs=cached_complexs
                 )
+                if use_only_cache:
+                    if inputs:
+                        logger.warning(
+                            f"Disregarding {len(inputs)} structures which could not be found in the cache."
```
Comment (hmacdope) [inline asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py:129]: Could be misunderstanding but isn't the message the reverse? Shouldn't it be something like 

```python
f"Disregarding inputs, {len(inputs)}, only loading structures prepped in cache."
```

**Review by hmacdope — CHANGES_REQUESTED** [review]
Looks good, just one query RE logging message.

File: asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py (hunk 1 above)
Comment (apayne97) [reply asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py:129]: Here is the message in context:

```
2023-12-14 11:49:03,683 | root | INFO | cross_docking.py | cross_docking_workflow | Loaded 219 complexes
2023-12-14 11:49:03,685 | root | INFO | cross_docking.py | cross_docking_workflow | Prepping complexes
2023-12-14 11:49:06,685 | asapdiscovery.modeling.protein_prep_v2 | INFO | protein_prep_v2.py | prep | Loaded 218 cached structures from: /lila/data/chodera/asap-datasets/full_frag_prepped_mpro_20230122/du_cache_p_only.
2023-12-14 11:49:09,801 | asapdiscovery.modeling.protein_prep_v2 | WARNING | protein_prep_v2.py | prep | Disregarding 3 structures which could not be found in the cache.
2023-12-14 11:49:09,808 | asapdiscovery.modeling.protein_prep_v2 | INFO | protein_prep_v2.py | prep | Matched 216 cached structures which will be reused.
2023-12-14 11:49:09,811 | root | INFO | cross_docking.py | cross_docking_workflow | Prepped 216 complexes
```

Comment (apayne97) [reply asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py:129]: I think this makes sense to me?

Comment (hmacdope) [reply asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py:129]: Yep makes sense in context, my confusion.

**Review by hmacdope — APPROVED** [review]
Thanks @apayne97

**Review by jthorton — APPROVED** [review]
This seems like a good workaround for now, I guess we could also have an option like `ignore-fails` for when you still want to try to prep everything first but that could be added separately.

**Pull request merged**